src/sqlfluff/core/linter/linter.py

```diff
@@ -0,0 +1,760 @@
+
+    @staticmethod
+    def _parse_tokens(
+        tokens: Sequence[BaseSegment], config: FluffConfig, recurse: bool = True
+    ) -> Tuple[Optional[BaseSegment], List[SQLParseError]]:
+        parser = Parser(config=config)
+        violations = []
+        # Parse the file and log any problems
+        try:
+            parsed: Optional[BaseSegment] = parser.parse(tokens, recurse=recurse)
+        except SQLParseError as err:
+            linter_logger.info("PARSING FAILED! : %s", err)
+            violations.append(err)
+            return None, violations
+
+        if parsed:
+            linter_logger.info("\n###\n#\n# {0}\n#\n###".format("Parsed Tree:"))
+            linter_logger.info("\n" + parsed.stringify())
+            # We may succeed parsing, but still have unparsable segments. Extract them here.
+            for unparsable in parsed.iter_unparsables():
+                # No exception has been raised explicitly, but we still create one here
+                # so that we can use the common interface
+                violations.append(
+                    SQLParseError(
+                        "Line {0[0]}, Position {0[1]}: Found unparsable section: {1!r}".format(
+                            unparsable.pos_marker.working_loc,
+                            unparsable.raw
+                            if len(unparsable.raw) < 40
+                            else unparsable.raw[:40] + "...",
+                        ),
+                        segment=unparsable,
+                    )
+                )
+                linter_logger.info("Found unparsable segment...")
+                linter_logger.info(unparsable.stringify())
+        return parsed, violations
+
+    @staticmethod
+    def _generate_short_fname(fname: Optional[str] = None):
+        # Handle nulls gracefully
+        if not fname:
+            return None
+        return fname.replace("\\", "/").split("/")[-1]
+
+    def render_string(self, in_str: str, fname: Optional[str], config: FluffConfig):
+        """Template the file."""
+        linter_logger.info("TEMPLATING RAW [%s] (%s)", self.templater.name, fname)
+
+        try:
+            templated_file, templater_violations = self.templater.process(
+                in_str=in_str, fname=fname, config=config, formatter=self.formatter
+            )
+        except SQLTemplaterSkipFile as s:
+            linter_logger.warning(str(s))
+            templated_file = None
+            templater_violations = []
+
+        if not templated_file:
+            linter_logger.info("TEMPLATING FAILED: %s", templater_violations)
+
+        return templated_file, templater_violations
+
+    def parse_string(
+        self,
+        in_str: str,
+        fname: Optional[str] = None,
+        recurse: bool = True,
+        config: Optional[FluffConfig] = None,
+    ) -> ParsedString:
+        """Parse a string.
+
+        Returns:
+            `ParsedString` of (`parsed`, `violations`, `time_dict`, `templated_file`).

```

Comment:
Now that `ParsedString` is defined as a class, could these comments be removed (or moved to `ParsedString` if not already documented there)?

Reply:
Yes. Will do.